rollup.config.js

```diff
@@ -1,34 +1,52 @@
-
-import buble from 'rollup-plugin-buble';
-import filesize from 'rollup-plugin-filesize';
-import resolve from 'rollup-plugin-node-resolve';
-import commonjs from 'rollup-plugin-commonjs';
-import uglify from 'rollup-plugin-uglify';
+import buble from "rollup-plugin-buble";
+import filesize from "rollup-plugin-filesize";
+import resolve from "rollup-plugin-node-resolve";
+import commonjs from "rollup-plugin-commonjs";
+import uglify from "rollup-plugin-uglify";
 import json from "rollup-plugin-json";
-import typescript from 'rollup-plugin-typescript';
+import typescript from "rollup-plugin-typescript";
+import replace from "rollup-plugin-replace";
+
+const MODULE_NAME = "AlgoliaAnalytics",
+  LIBRARY_OUTPUT_NAME = "search-insights";
 
-const MODULE_NAME = 'AlgoliaAnalytics', LIBRARY_OUTPUT_NAME = 'search-insights';
+const createPlugins = ({ isBrowserBuild }) => [
+  typescript(),
+  replace({
+    __BROWSER_BUILD__: isBrowserBuild
+  }),
+  resolve({
+    preferBuiltins: false
+  }),
+  json({
+    preferConst: true,
+    compact: true
+  }),
+  buble(),
+  commonjs(),
+  uglify(),
+  filesize()
+];
 
-export default {
-  entry: 'lib/insights.ts',
-  format: 'umd',
-  moduleName: MODULE_NAME,
-  plugins: [
-    typescript(),
-    resolve({
-      browser: true,
-      preferBuiltins: false,
-    }),
-    json({
-      preferConst: true,
-      compact: true
-    }),
-    buble(),
-    commonjs(),
-    uglify(),
-    filesize()
-  ],
-  targets: [
-    { dest: `./dist/${LIBRARY_OUTPUT_NAME}.min.js`, format: 'umd' },
-  ],
-};
+export default [
+  {
+    input: "lib/insights.ts",
+    output: {
+      format: "umd",
+      name: MODULE_NAME,
+      file: `./dist/${LIBRARY_OUTPUT_NAME}.min.js`,
+      globals: {}
+    },
+    plugins: createPlugins({ isBrowserBuild: true })
+  },
+  {
+    input: "lib/insights.ts",
+    output: {
+      format: "cjs",
```

Comment:
If we go with two entry points we could get rid of the `cjs` Rollup build. We can transpile the code with TypeScript (or Babel/buble depends on the setup) and ship it as is, like a regular Node lib.

Reply:
@samouss 
> We can transpile the code with TypeScript (or Babel/buble depends on the setup) and ship it as is, like a regular Node lib.

To make sure if I understand correctly, this means we'll add `prepare` script to transpile the code with TypeScript and the `main` points to something like `dist/index.js`, right?

Reply:
The `prepare` step is not mandatory. You can take a look at InstantSearch.js or React InstantSearch this is how it's done. Basically the `cjs` environment does require a flat bundle since most of the time there is a module loader/bundler that does it for you.

Reply:
Thanks. I will do this part in a separate PR right after I finish the node compatibility PRs.